code/javascript/example.js

```diff
@@ -1,27 +1,41 @@
 // npm install --save neo4j-driver
 // node example.js
-const neo4j = require('neo4j-driver');
-const driver = neo4j.driver('neo4j+s://demo.neo4jlabs.com:7687',
-                  neo4j.auth.basic('mUser', 's3cr3t'), 
-                  {});
+const neo4j = require("neo4j-driver");
 
-const query =
-  `
-  MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person) RETURN a.name as actorName
-  `;
+async function main() {
+  const driver = neo4j.driver("neo4j+s://demo.neo4jlabs.com:7687", neo4j.auth.basic("mUser", "s3cr3t"));
 
-const params = {"movieTitle": "The Matrix"};
+  try {
+    const products = await getDairyProducts(driver)
+    products.forEach(product => console.log(product))
+  } catch (error) {
+    console.error(error);
+  } finally {
+    await driver.close()
+  }
+}
+
+async function getDairyProducts(driver) {
+  const session = driver.session({ database: "movies" });
+
+  try {
+    return await session.readTransaction(async tx => {
+      const query = `
+        MATCH (m:Movie {title:$movieTitle})<-[:ACTED_IN]-(a:Person) 
+        RETURN a.name as actorName
+        `;
 
-const session = driver.session({database:"movies"});
+      const params = { movieTitle: "The Matrix" };
 
-session.run(query, params)
-  .then((result) => {
-    result.records.forEach((record) => {
-        console.log(record.get('actorName'));
+      const result = await tx.run(query, params);
+
+      return result.records.map(record => record.get('actorName'))
     });
-    session.close();
-    driver.close();
-  })
-  .catch((error) => {
-    console.error(error);
-  });
+  } finally {
+    await session.close();
+  }
+}
+
+main()
```

Comment:
I wasn't aware that we need a main in JS?
can we avoid this?

Reply:
This issue is directed linked with: https://github.com/neo4j-graph-examples/template/pull/31/files#r949406759

We don't need `main` in the JS, but we could not `await` in the top level until `Node 14.8.0`. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await

There is two common techniques for solving this:
1. Create an `async function main()` (or other arbitrary name), implement everything using `async/await` inside `main` and then call the function (and tread the promise completion). 
2. Mostly the same, but with an anonymous function. Like:
```javascript
(async function () { 
   // the async/ await stuff
})()
  .then(() => console.log('finished'))
  .catch(err => console.error(err)) // or do a proper error treatment.
```

We could also revert for using only promises, which doesn't look great. Or we could write `top-level` await considering the user will be running newer Node versions (which I prefer)

Reply:
We'll leave it as it is then.